modules/phase_field/src/kernels/ADSplitCHBase.C

```diff
@@ -0,0 +1,37 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADSplitCHBase.h"
+
+defineADValidParams(ADSplitCHBase,
+                    ADKernel,
+                    params.addClassDescription("Base class for split Cahn-Hilliard equation."););
+
+template <ComputeStage compute_stage>
+ADSplitCHBase<compute_stage>::ADSplitCHBase(const InputParameters & parameters)
+  : ADKernel<compute_stage>(parameters)
+{
+}
+
+template <ComputeStage compute_stage>
+ADResidual
+ADSplitCHBase<compute_stage>::computeQpResidual()
+{
+  return computeDFDC() * _test[_i][_qp];
```

Comment:
I'll ask one more time and then stop bugging you about it...candidate for `ADKernelValue`? The reason I ask these questions is because I worry about future complaints about the speed of AD MOOSE.

One thing I would like to do pretty quickly is to use regular `_coord`, `_JxW`, and perhaps even `_grad_test` when not calculating on the displaced mesh

Reply:
ADSplitCHCRes inherits from this and adds a term with `_grad_test` onto its residual.

Reply:
The non-AD `_coord` etc. for undisplaced meshes sounds like a good idea though.